### ISSUE TITLE
fatal error: 'mobileffmpeg/LogDelegate.h' file not found

### ISSUE THREAD

**Issue by abahnj** [tanersener/flutter-ffmpeg#309]
**Description**
This error occurs when trying to build for iOS. 
duplicate of #303 

**Expected behavior**
That the build succeeds successfully and xcframework files and headers are linked

**Current behavior**
The build fails

**Screenshots**

**Logs**
Below I've pasted all log outputs that has to do with flutter ffmpeg but before that you'll find a ghostbin link that includes the other log content

https://ghostbin.co/paste/59b8y

  CompileC /Users/abahnj/Library/Developer/Xcode/DerivedData/Runner-ftlsjxynvnhlikesdqqjblzxnjjq/Build/Intermediates.noindex/Pods.build/Debug-Development-iphonesimulator/Reachability.build/Objects-normal/x86_64/Reachability.o
    /Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/Reachability/Reachability.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
    error: the following command failed with exit code 0 but produced no further output
    CompileC /Users/abahnj/Library/Developer/Xcode/DerivedData/Runner-ftlsjxynvnhlikesdqqjblzxnjjq/Build/Intermediates.noindex/Pods.build/Debug-Development-iphonesimulator/Reachability.build/Objects-normal/arm64/Reachability.o
    /Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/Reachability/Reachability.m normal arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler

 warning: [CP] Unable to find matching .xcframework slice in '/Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/mobile-ffmpeg-min/mobileffmpeg.xcframework mobileffmpeg framework ios-arm64 ios-x86_64-simulator ios-x86_64-maccatalyst' for the current build architectures (arm64 x86_64).
    warning: [CP] Unable to find matching .xcframework slice in '/Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/mobile-ffmpeg-min/libavcodec.xcframework libavcodec framework ios-x86_64-simulator ios-x86_64-maccatalyst ios-arm64' for the current build architectures (arm64 x86_64).
    warning: [CP] Unable to find matching .xcframework slice in '/Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/mobile-ffmpeg-min/libavdevice.xcframework libavdevice framework ios-x86_64-simulator ios-x86_64-maccatalyst ios-arm64' for the current build architectures (arm64 x86_64).
    warning: [CP] Unable to find matching .xcframework slice in '/Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/mobile-ffmpeg-min/libavfilter.xcframework libavfilter framework ios-x86_64-maccatalyst ios-arm64 ios-x86_64-simulator' for the current build architectures (arm64 x86_64).
    warning: [CP] Unable to find matching .xcframework slice in '/Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/mobile-ffmpeg-min/libavformat.xcframework libavformat framework ios-arm64 ios-x86_64-maccatalyst ios-x86_64-simulator' for the current build architectures (arm64 x86_64).
    warning: [CP] Unable to find matching .xcframework slice in '/Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/mobile-ffmpeg-min/libavutil.xcframework libavutil framework ios-arm64 ios-x86_64-maccatalyst ios-x86_64-simulator' for the current build architectures (arm64 x86_64).
    warning: [CP] Unable to find matching .xcframework slice in '/Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/mobile-ffmpeg-min/libswresample.xcframework libswresample framework ios-x86_64-maccatalyst ios-arm64 ios-x86_64-simulator' for the current build architectures (arm64 x86_64).
    warning: [CP] Unable to find matching .xcframework slice in '/Users/abahnj/AndroidStudioProjects/oyoyo/ios/Pods/mobile-ffmpeg-min/libswscale.xcframework libswscale framework ios-arm64 ios-x86_64-maccatalyst ios-x86_64-simulator' for the current build architectures (arm64 x86_64).

******

1 warning generated.
    In file included from /Users/abahnj/.pub-cache/hosted/pub.dartlang.org/flutter_ffmpeg-0.4.0/ios/Classes/EmptyLogDelegate.m:20:
    /Users/abahnj/.pub-cache/hosted/pub.dartlang.org/flutter_ffmpeg-0.4.0/ios/Classes/EmptyLogDelegate.h:20:10: fatal error: 'mobileffmpeg/LogDelegate.h' file not found
    #include <mobileffmpeg/LogDelegate.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

**Environment**
Flutter (Channel stable, 2.2.1, on macOS 11.4 20F71 darwin-x64, locale en-GB)
    • Flutter version 2.2.1 at /Users/abahnj/Developer/flutter
    • Framework revision 02c026b03c (4 days ago), 2021-05-27 12:24:44 -0700
    • Engine revision 0fdb562ac8
    • Dart version 2.13.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/abahnj/Library/Android/sdk
    • Platform android-S, build-tools 30.0.3
    • ANDROID_HOME = /Users/abahnj/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (4 available)
    • SM N960U1 (mobile) • 25abfa8e08017ece                     • android-arm64  • Android 10 (API 29)
    • iPhone 12 (mobile) • 38B6E151-E2D4-434A-B7DB-DA5FEF02ABC3 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64     • macOS 11.4 20F71 darwin-x64
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 90.0.4430.212
    ! Error: Abah’s iPhone is not connected. Xcode will continue when Abah’s iPhone is connected. (code -13)

**Other**
One thing I've noticed in trying to troubleshoot this problem is that switching to flutter 2.0.6 always successfully builds and then switching back to 2.2 the build is successful as long as you don't perform a flutter clean or restart your emulator.

another detail that might be important is that I have multiple flavors setup in the iOS project

Also I noticed that running the app straight from xcode works

**Comment by fauzipadlaw** [comment]
same issue:

```
  1 error generated.
    In file included from /Users/efzet/.pub-cache/git/flutter-ffmpeg-59823917461dd67a1ad1eee8a254b283fbd95bac/ios/Classes/EmptyLogDelegate.m:20:
    /Users/efzet/.pub-cache/git/flutter-ffmpeg-59823917461dd67a1ad1eee8a254b283fbd95bac/ios/Classes/EmptyLogDelegate.h:20:10: fatal error: 'mobileffmpeg/LogDelegate.h' file not found
    #include <mobileffmpeg/LogDelegate.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete
```

flutter doctor:
```
[✓] Flutter (Channel beta, 2.2.0, on macOS 11.4 20F71 darwin-arm, locale en-ID)
    • Flutter version 2.2.0 at /Users/efzet/SDKs/flutter
    • Framework revision b22742018b (2 weeks ago), 2021-05-14 19:12:57 -0700
    • Engine revision a9d88a4d18
    • Dart version 2.13.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/efzet/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/efzet/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave
      Browser

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.3.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 56.0.3
    • Dart plugin version 203.8173

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 34465855-802A-4D1E-AF3E-9F0FFC936150 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)               • chrome                               •
      web-javascript • Brave Browser 91.1.25.68

• No issues found!
```

**Comment by fauzipadlaw** [comment]
my Podfile:

```
# Uncomment this line to define a global platform for your project
platform :ios, '11.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

def flutter_install_ios_plugin_pods(ios_application_path = nil)
  # defined_in_file is set by CocoaPods and is a Pathname to the Podfile.
  ios_application_path ||= File.dirname(defined_in_file.realpath) if self.respond_to?(:defined_in_file)
  raise 'Could not find iOS application path' unless ios_application_path

  # Prepare symlinks folder. We use symlinks to avoid having Podfile.lock
  # referring to absolute paths on developers' machines.

  symlink_dir = File.expand_path('.symlinks', ios_application_path)
  system('rm', '-rf', symlink_dir) # Avoid the complication of dependencies like FileUtils.

  symlink_plugins_dir = File.expand_path('plugins', symlink_dir)
  system('mkdir', '-p', symlink_plugins_dir)

  plugins_file = File.join(ios_application_path, '..', '.flutter-plugins-dependencies')
  plugin_pods = flutter_parse_plugins_file(plugins_file)
  plugin_pods.each do |plugin_hash|
    plugin_name = plugin_hash['name']
    plugin_path = plugin_hash['path']
    if (plugin_name && plugin_path)
      symlink = File.join(symlink_plugins_dir, plugin_name)
      File.symlink(plugin_path, symlink)

      if plugin_name == 'flutter_ffmpeg'
          pod 'flutter_ffmpeg/full-gpl-lts', :path => File.join(relative_symlink_dir, 'plugins', plugin_name, platform)
      else
          pod plugin_name, :path => File.join(relative_symlink_dir, 'plugins', plugin_name, platform)
      end
    end
  end
end

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '11.0'
    end
  end
end
```

**Comment by tanersener** [comment]
Could you explain on which devices you get this error? Only on `x64` emulator?

**Comment by tanersener** [comment]
`logs` in your post tell that there is a mismatch between architectures your app wants and architectures `flutter_ffmpeg` provides. Have you tried running the test application under the `example` folder?

**Comment by abahnj** [comment]
Not just on an x64 emulator - I'm assuming the default emulators included with xcode are x64 - but only on a physical iPhone 8 plus which is the only physical iOS device I have available. I have tried running the test application in the example folder and it runs successfully which is what really stumps me. A difference i noticed in the example podfile and mine is the use_frameworks! and use_modular_headers! tag

**Comment by abahnj** [comment]
So a couple of things I realized.
I realized that changing from min to min-lts fixed the issue for me. So maybe try other versions of the package
 Also if none of your dependencies require use_frameworks! and/or use_modular_headers! remove them one at a time.

**Comment by surielb** [comment]
This fixed it for me:

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)

    # Add the following:
    target.build_configurations.each do |build_configuration|
      build_configuration.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = 'arm64 i386'
    end
  end
end
```

[taken from here](https://github.com/flutter/flutter/issues/64502#issuecomment-759165857)

**Comment by lvlrSajjad** [comment]
> This fixed it for me:
> 
> ```
> post_install do |installer|
>   installer.pods_project.targets.each do |target|
>     flutter_additional_ios_build_settings(target)
> 
>     # Add the following:
>     target.build_configurations.each do |build_configuration|
>       build_configuration.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = 'arm64 i386'
>     end
>   end
> end
> ```
> 
> [taken from here](https://github.com/flutter/flutter/issues/64502#issuecomment-759165857)

Didn't work for me.

**Comment by ah89** [comment]
> This fixed it for me:
> 
> ```
> post_install do |installer|
>   installer.pods_project.targets.each do |target|
>     flutter_additional_ios_build_settings(target)
> 
>     # Add the following:
>     target.build_configurations.each do |build_configuration|
>       build_configuration.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = 'arm64 i386'
>     end
>   end
> end
> ```
> 
> [taken from here](https://github.com/flutter/flutter/issues/64502#issuecomment-759165857)

Thanks its worked

**Comment by SinoMiles** [comment]
target 'Runner' do
  #use_frameworks! **//remove #**
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
